app/models/user.rb

```diff
@@ -19,7 +24,13 @@ def social_worker?
     role == User::SOCIAL_WORKER
   end
 
-  validates :role, inclusion: { in: ROLES }
+  def department_manager?
+    role == User::DEPARTMENT_MANAGER
+  end
+
+  def superadmin_or_department_manager?
```

Comment:
Don't repeat yourself is a good advice, but the goal is to have readable, extensible and maintainable code.

And as every pattern, they are advices, not rules